src/main/java/repository/Repository.java

```diff
@@ -0,0 +1,18 @@
+package repository;
+
+import java.nio.file.Path;
+import java.util.List;
+
+public interface Repository {
+    List<String> readDirectoryContents(String path);
+    byte[] readFile(String path);
+    String getFileType(String path);
+    void writeFile(String intendedFilePath, String fileType, byte[] fileContents);
+    void createDirectories(Path directoryPaths);
+    void deleteDirectory(String directoryPath);
+    void deleteFile(String filePath);
+    static String trimLastResource(String path) {
```

Comment:
Do you think this belongs in the interface?

Reply:
It does not and i've removed a few other unnecessary interface methods